spring-integration-core/src/main/java/org/springframework/integration/support/converter/ObjectStringMessageConverter.java

```diff
@@ -24,14 +26,30 @@
  * <p>
  * Delegates to super when payload is {@code byte[]} or {@code String}.
  * Performs {@link Object#toString()} in other cases.
+ * <p>
+ * This class meant to be used as a fallback converter internally when deserializing messages. Therefore, only
+ * {@link org.springframework.messaging.converter.AbstractMessageConverter#fromMessage(Message, Class) fromMessage}
+ * method should be called with {@code String.class} as {@code targetClass}, obviously.To be explicit, using method
```

Comment:
Sorry, I don't find words like `obviously` and `To be explicit` as formal technical documentation language.
I also don't this that we need to mention `toMessage()` in this Javadoc at all.
It is really the superclass responsibility to express its nature.
we override only `convertFromInternal()` and we should explain for what reason.
Nothing more to make the reader to feel silly.